pkg/handlers/generic/lifecycle/utils/utils.go

```diff
@@ -86,3 +107,95 @@ func EnsureNamespace(ctx context.Context, c ctrlclient.Client, name string) erro
 
 	return nil
 }
+
+func RetrieveValuesTemplateConfigMap(
+	ctx context.Context,
+	c ctrlclient.Client,
+	configMapName,
+	defaultsNamespace string,
+) (*corev1.ConfigMap, error) {
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: defaultsNamespace,
+			Name:      configMapName,
+		},
+	}
+	configMapObjName := ctrlclient.ObjectKeyFromObject(
+		configMap,
+	)
+	err := c.Get(ctx, configMapObjName, configMap)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"failed to retrieve installation values template ConfigMap %q: %w",
+			configMapObjName,
+			err,
+		)
+	}
+	return configMap, nil
+}
+
+func CreateStorageClass(
+	ctx context.Context,
+	cl ctrlclient.Client,
+	storageConfig v1alpha1.StorageClassConfig,
+	cluster *clusterv1.Cluster,
+	defaultsNamespace,
+	provisionerName string,
+	isDefault bool,
+) error {
+	var volumeBindingMode *storagev1.VolumeBindingMode
+	switch storageConfig.VolumeBindingMode {
+	case v1alpha1.VolumeBindingImmediate:
+		volumeBindingMode = ptr.To(storagev1.VolumeBindingImmediate)
+	case v1alpha1.VolumeBindingWaitForFirstConsumer:
+	default:
+		volumeBindingMode = ptr.To(storagev1.VolumeBindingWaitForFirstConsumer)
+	}
+	var reclaimPolicy *corev1.PersistentVolumeReclaimPolicy
+	switch storageConfig.ReclaimPolicy {
+	case v1alpha1.VolumeReclaimRecycle:
+		reclaimPolicy = ptr.To(corev1.PersistentVolumeReclaimRecycle)
+	case v1alpha1.VolumeReclaimDelete:
+		reclaimPolicy = ptr.To(corev1.PersistentVolumeReclaimDelete)
+	case v1alpha1.VolumeReclaimRetain:
+		reclaimPolicy = ptr.To(corev1.PersistentVolumeReclaimRetain)
+	}
+	params := defaultParams
+	if storageConfig.Parameters != nil {
```

Comment:
Are we missing some required parameters in the SC? https://github.com/nutanix/helm/blob/4ce73556497c851d1e17f197ee96f99b773c49f8/charts/nutanix-csi-storage/templates/ntnx-sc.yaml#L22-L29